include/drivers/mdio.h

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file
+ *
+ * @brief Public APIs for MDIO drivers.
+ */
+
+/*
+ * Copyright (c) 2021 IP-Logix Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DRIVERS_MDIO_H_
+#define ZEPHYR_INCLUDE_DRIVERS_MDIO_H_
+
+/**
+ * @brief MDIO Interface
+ * @defgroup mdio_interface MDIO Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+#include <zephyr/types.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @cond INTERNAL_HIDDEN
+ *
+ * These are for internal use only, so skip these in
+ * public documentation.
+ */
+__subsystem struct mdio_driver_api {
```

Comment:
Why `__subsystem` and `__syscalls` below? These should only needed if you want to call these functions from userspace when `CONFIG_USERSPACE` is enabled, not sure what would be the use case for doing that.